src/screens/augmentedReality/ARShowScreen.js

```diff
@@ -163,17 +163,21 @@ export const ARShowScreen = ({ navigation, route }) => {
 };
 
 const objectParser = async ({ item, setObject, setIsLoading, onPress }) => {
-  let parsedObject = {};
+  const parsedObject = { texture: [] };
 
-  if (item.animationName) {
-    parsedObject.animationName = item.animationName;
+  if (item?.payload?.animationName) {
+    parsedObject.animationName = item?.payload?.animationName;
   }
 
-  item?.localUris?.forEach((item) => {
-    parsedObject[item.type] = item.uri;
+  item?.payload?.localUris?.forEach((item) => {
+    if (item.type === 'texture') {
+      parsedObject[item.type].push({ uri: item.uri });
+    } else {
+      parsedObject[item.type] = item.uri;
+    }
   });
 
-  if (!parsedObject.png || !parsedObject.vrx) {
+  if (!parsedObject?.texture?.length || !parsedObject?.vrx) {
```

Comment:
with creating `const parsedObject = { texture: [] };` the check of `!parsedObject.texture` would be always false, so I added length check for array

Reply:
Thank you 🙏